lib/routing_filter/versioned_path.rb

```diff
@@ -0,0 +1,49 @@
+# TODO: Remove this when v1 is dead
+module RoutingFilter
+  class VersionedPath < Filter
+    HEADER_VERSION_REGEX = %r{.*version=(?<version>\d+)}.freeze
+    PATH_VERSION_REGEX = %r{\A\/api(\/(?<version>v1))\/.+\Z}.freeze
+
+    def around_recognize(path, env)
+      raise ActionController::RoutingError, 'Not Found' if raise_not_found_error?(path, env)
```

Comment:
Remove this for backwards compatibility for now. We're going to hard drop support for v1 in the path once the supplier has migrated over to v2

Reply:
Speaking to Martyn and this is necessary.